src/ZoneServer/Skills/Handlers/Swordsman/PommelBeat.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.L10N;
+using Melia.Shared.Game.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using static Melia.Zone.Skills.SkillUseFunctions;
+using Melia.Zone.World.Actors.Monsters;
+
+namespace Melia.Zone.Skills.Handlers.Swordsman
+{
+	/// <summary>
+	/// Handler for the Swordsman skill Pommel Beat.
+	/// </summary>
+	[SkillHandler(SkillId.Swordman_PommelBeat)]
+	public class PommelBeat : IGroundSkillHandler
+	{
+		private const float StunDamageMultiplier = 3f; // 300% damage if target is stunned
+		private const float BaseIgnoreDefRate = 0.01f; // 1% ignore defense
+		private const float IgnoreDefRatePerLevel = 0.005f; // 0.5% ignore defense per level
+
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(farPos);
+			caster.SetAttackState(true);
+
+			var splashParam = skill.GetSplashParameters(caster, originPos, farPos, length: 50, width: 25, angle: 0);
+			var splashArea = skill.GetSplashArea(SplashType.Square, splashParam);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, farPos, null);
+
+			this.Attack(skill, caster, splashArea);
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(300);
+			var damageDelay = TimeSpan.FromMilliseconds(150);
+			var skillHitDelay = TimeSpan.Zero;
+
+			await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var hits = new List<SkillHitInfo>();	
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				SkillModifier modifier = new SkillModifier();
+
+				if (target.IsBuffActive(BuffId.Stun))
+				{
+					modifier.DamageMultiplier = StunDamageMultiplier;
+				}
+				var targetSize = SizeType.M; // non-mobs are always size M
+				if (target is Mob mob)
```

Comment:
With the goal of making skill handlers as target type agnostic as possible in mind, it might be better to create an extension methods that handles this, returning either the mob data or Mas the default for a combat entity. Or whatever else might be necessary, in case there's a way for players to change their size.

Reply:
Another thing to potentially keep in mind, which I just learned about, is that players actually aren't size `M`. They have their own size, called `PC`. Though in certain situations they are equivalent to size `M`, as these two share a "size rank". That's presumably what we want to query for at some point in the future. See `shape.ies` for some more information.

Reply:
So should we create a SizeType.PC for players to have?  I feel like maybe it's best to leave the implementation of this one up to you.

Reply:
Looking at the file mentioned, we probably do want a PC size, but yea, we/I can look into that another time.